postgres/configure-and-start-postgres.sh

```diff
@@ -0,0 +1,142 @@
+while [ $# -gt 0 ]; do
+    if [ "${1}" = "--host" ]; then
+        shift; if [ -z "${1}" ]; then usage; fi
+        HOST="${1}"
+        shift; continue
+    fi
+    if [ "${1}" = "--dbname" ]; then
+        shift; if [ -z "${1}" ]; then usage; fi
+        DBNAME="${1}"
+        shift; continue
+    fi
+    if [ "${1}" = "--username" ]; then
+        shift; if [ -z "${1}" ]; then usage; fi
+        USERNAME="${1}"
+        shift; continue
+    fi
+    if [ "${1}" = "--password" ]; then
+        shift; if [ -z "${1}" ]; then usage; fi
+        PASSWORD="${1}"
+        shift; continue
+    fi
+    if [ "${1}" = "--data-dir" ]; then
+        shift; if [ -z "${1}" ]; then usage; fi
+        DATA_DIR="${1}"
+        shift; continue
+    fi
+    echo -e "\nParameter '$1' is not recognized\n"
+    usage
+done
+
+if [ -z "${HOST}" ] || [ -z "${DBNAME}" ] || [ -z "${USERNAME}" ] || [ -z "${PASSWORD}" ]; then
+    usage
+fi
+
+
+#####################################
+### Actual installation procedure ###
+#####################################
+
+${SCRIPT_DIR}/install.sh
+
+log "Updating Postgres configs to accept host connections to the Database"
+PG_MAJOR_VER=$(psql --version | awk '{ print $3 }' | awk -F '.' '{ print $1 }')
+PG_VERSION=$(psql --version | awk '{ print $3 }' | awk -F '.' '{ print $1 "." $2 }')
+
+if [ "${PG_MAJOR_VER}" = "10" ]; then
+    PG_CONFIG_DIR="/etc/postgresql/${PG_MAJOR_VER}"
+else
+    PG_CONFIG_DIR="/etc/postgresql/${PG_VERSION}"
+fi
+
+PG_CONF="${PG_CONFIG_DIR}/main/postgresql.conf"
+PG_HBA="${PG_CONFIG_DIR}/main/pg_hba.conf"
+
+# Configure postgres to listen on the specified host
+if grep -q -E "^listen_addresses *=" "${PG_CONF}"
+then
+    sed -i "s/^listen_addresses *=.*/listen_addresses = 'localhost,${HOST}'/" "${PG_CONF}"
+else
+    echo "listen_addresses = 'localhost,${HOST}'" >> "${PG_CONF}"
+fi
+
+# Configure postgres to use specific data directory
+if [ -n "${DATA_DIR}" ]; then
+    if grep -q -E "^data_directory *=" "${PG_CONF}"
+    then
+        sed -i "s/^data_directory *=.*/data_directory = '${DATA_DIR}'/" "${PG_CONF}"
+    else
+        echo "data_directory = '${DATA_DIR}'" >> "${PG_CONF}"
+    fi
+fi
+
+# Allow host connections to the specified DB
+if grep -q -E "^host[ \t]+${DBNAME}[ \t]+${USERNAME}[ \t]+" "${PG_HBA}"
+then
+    sed -i "s|^host[ \t]+${DBNAME}[ \t]+${USERNAME}[ \t]+.*|host ${DBNAME} ${USERNAME} ${HOST}/0 md5|" "${PG_HBA}"
+else
+    echo "host ${DBNAME} ${USERNAME} ${HOST}/0 md5" >> "${PG_HBA}"
+fi
+
+systemctl restart postgresql.service
+systemctl enable postgresql.service
+systemctl status postgresql.service
+
+
+echo "${HOST}:5432:${DBNAME}:${USERNAME}:${PASSWORD}" > ~/.pgpass
+chmod 600 ~/.pgpass
+
+log "Checking if DB and user already exist"
+if psql --dbname ${DBNAME} --username ${USERNAME} --host ${HOST} \
+        --command 'SELECT current_timestamp;'
+then
+    rm ~/.pgpass
+    log "DB and user are already configured"
+    exit 0
+else
+    rm ~/.pgpass
+fi
```

Comment:
You can avoid password files and using sudo by making use of `pg_ident.conf` to map `root` to `postgres`.

in `pg_ident.conf`: 
`rootmap         root                    postgres`

update `pg_hba.conf`: 
`local   all             postgres                                peer map=rootmap`

then access as postgres using `psql -U postgres` or `createdb -U postgres`.

Reply:
Thanks, I didn't know about `rootmap` feature.
The reason why `.pgpass` file is used is to test if specified database was previously created and specified user can login with provided password (so to test that user will be able to connect using corresponding DSN string).
I see that `rootmap` would also help to get rid of `sudo -u`, but I'd prefer to leave it as it is for now.
Let me know if you think it's critical.